ref(grouping): change GroupingEventVariantType to use underscores in FE

shayna-ch opened the pull request:
Changes `-` to `_` in `eventGroupVariantType` in `event.tsx`. 

Since data is sent with dashes in the back end, `convertVariantFromBackend()` is needed to convert incoming data from back end with dashes to data with underscores in the front end. This function will eventually be removed when the back end is changed to send data in the correct format. 

*First PR in a series of changes to change - to _ in eventGroupVariantType for consistency across front end and back end.*

shayna-ch committed:
ref(grouping): change GroupingEventVariantType to use underscores in FE

shayna-ch committed:
ref(grouping): change GroupingEventVariantType to use underscores in FE

shayna-ch committed:
ref(grouping): change GroupingEventVariantType to use underscores in FE

shayna-ch committed:
fix export

shayna-ch committed:
type check for variant

shayna-ch committed:
add check for eventGroupVariantType before converting

shayna-ch committed:
add check for eventGroupVariantType before converting

shayna-ch committed:
add check for eventGroupVariantType before converting

shayna-ch committed:
fix isEventGroupVariantType to use dashes

lobsterkatie reviewed (commented):
Overall approach looks good! Just a few small questions.

shayna-ch committed:
Update static/app/types/event.tsx

Co-authored-by: Katie Byers <katie.byers@sentry.io>

shayna-ch committed:
Merge branch 'master' into shayna-ch/make-FE-accept-underscores

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes

isabellaenriquez committed:
feat(checkout v3): Update step header styling (#97680)

Closes
https://linear.app/getsentry/issue/BIL-1201/update-checkout-step-component
Figma:
https://www.figma.com/design/4ICnO5MHVqkm6JV0TKzf7H/Checkout-Redesign?node-id=2682-35410&t=HhE5uBzZqURJKr0h-1

This just deals with updating the step header styling, not building out
the new steps. Per discussion with @itsdangold, we will not add
collapsible steps right now. [We will also not be porting over the
legacy
toggle](https://sentry.slack.com/archives/C08BMQJ2VFE/p1754683225769579?thread_ts=1754680793.775069&cid=C08BMQJ2VFE).

<img width="268" height="76" alt="Screenshot 2025-08-12 at 11 45 42 AM"
src="https://github.com/user-attachments/assets/c56b0fe1-8f34-4435-bae7-c55642b24e74"
/>

shayna-ch committed:
fix imports

ajay-sentry committed:
chore(prevent): add prevent ai flag to temp flag options (#97617)

This PR adds a new ff to the temp options for the prevent AI UI

relates to
https://github.com/getsentry/sentry-options-automator/pull/4883

<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

getsantry[bot] committed:
:hammer_and_wrench: apply pre-commit fixes
